Purchases/Public/CustomerInfo.swift

```diff
@@ -158,17 +158,37 @@ import Foundation
                       transactionsFactory: TransactionsFactory())
     }
 
+    // swiftlint:disable function_body_length
     init(data: [String: Any], dateFormatter: DateFormatter, transactionsFactory: TransactionsFactory) throws {
+    // swiftlint:enable function_body_length
         guard let subscriberObject = data["subscriber"] as? [String: Any] else {
             Logger.error(Strings.customerInfo.missing_json_object_instantiation_error(maybeJsonData: data))
             throw CustomerInfoError.missingJsonObject
         }
 
-        guard let subscriberData = SubscriberData(subscriberData: subscriberObject,
-                                                  dateFormatter: dateFormatter,
-                                                  transactionsFactory: transactionsFactory) else {
+        let subscriberData: SubscriberData
+        do {
+             try subscriberData = SubscriberData(subscriberData: subscriberObject,
+                                                 dateFormatter: dateFormatter,
+                                                 transactionsFactory: transactionsFactory)
+        } catch let subscriberDataError {
             Logger.error(Strings.customerInfo.cant_instantiate_from_json_object(maybeJsonObject: subscriberObject))
-            throw CustomerInfoError.cantInstantiateJsonObject
+
+            guard let subscriberDataError = subscriberDataError as? SubscriberData.SubscriberDataError else {
+                throw CustomerInfoError.cantInstantiateJsonObject.addingUnderlyingError(subscriberDataError)
+            }
+
+            let extraContext: String?
+            switch subscriberDataError {
+            case .originalAppUserIdMissing:
+                let originalAppUserId = subscriberObject["original_app_user_id"] as? String ?? "missing"
+                extraContext = "original_app_user_id is: \(originalAppUserId)"
+            case .firstSeenMissing, .firstSeenFormat:
+                let firstSeen = subscriberObject["first_seen"] as? String ?? "missing"
+                extraContext = "first_seen is: \(firstSeen)"
+            }
```

Comment:
we could extract this into a method so we don't have to disable the swiftlint warning

Reply:
Good idea! Done! ✅